File: EmojibleIOS/EmojibleIOS/Models/TeacherModel.swift
//
//  TeacherModel.swift
//  EmojibleIOS
//
//  Created by Haldun Balim on 12.11.2020.
//

import Foundation
class TeacherModel: UserModel{
    override init(email:String, name:String, surname:String, birthDate:Date) {
        super.init(email:email, name:name, surname:surname, birthDate: birthDate)
    }
    
    override init(dictionary: [String:Any]) {
        super.init(dictionary: dictionary)
    }
    
    override var dictionary: [String : Any]{
        var dict = super.dictionary
        dict["userType"] = "Teacher"
        return dict
    }
}

File: EmojibleAndroid/app/src/main/java/com/dji/emojibleandroid/models/TeacherModel.kt
package com.dji.emojibleandroid.models

import android.os.Build
import androidx.annotation.RequiresApi
import java.time.LocalDate

class TeacherModel(
    email: String,
    name: String,
    surname: String,
    birthDate: String,
    var classIds: MutableList<String>? = mutableListOf()
) : UserModel(email, name, surname, birthDate) {
    constructor(dictionary: HashMap<String, Any>) : this(
        dictionary["email"] as String,
        dictionary["name"] as String,
        dictionary["surname"] as String,
        dictionary["birthDate"] as String
    ) {
        classIds = dictionary["classIds"] as MutableList<String>?
    }

    override var dictionary: HashMap<String, Any> = super.dictionary.let {
        it["classIds"] = classIds as Any
        it["userType"] = "Teacher"
        it
    }
}